### PR TITLE
Add the media migration scripts

### PR DESCRIPTION
**Pull request by ngilain**
[OO-MISC] Migration scripts to explain in the changelog (main configuration changes + migration configuration)

### CODE REVIEW

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+    );
+    const ALTERNATIVES_RENAMING = array(
+        'max_height' => 'fixed_height',
+        'max_width' => 'fixed_width'
+    );
+
+    private $container;
+    protected $configuration;
+
+    public function setContainer(ContainerInterface $container = null)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: phpdoc

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to update the nodes by redefining the media id storage
+     *
+     * @param array blockType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getUpdateNodesRequest($blockType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: getUpdateNodesRequest(array $blockType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to update the media format in bbcode located in tinymce from contents
+     *
+     * @param array contentType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getUpdateTinyMceInContentRequest($contentType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getUpdateTinyMceInContentRequest(array $contentType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 4]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to update the contents by redefining the media id storage
+     *
+     * @param array contentType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getUpdateContentsRequest($contentType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getUpdateContentsRequest(array $contentType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 5]
```diff
@@ -0,0 +1,779 @@
...
+                                    if (typeof mediaAttribute[j] == "string") {
+                                        mediaAttribute[j] = formatMediaId(mediaAttribute[j]);
+                                    }
+                                }
+                            }
+                        }';
+        }
+
+        $query .= '
+                        blocks[i]["attributes"]["' . $attributeName . '"] = mediaAttribute;
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: if for any reason the variable attributes is empty, the attributeName doesn't exist

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 6]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to update the tinymce used in nodes
+     *
+     * @param array blockType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getUpdateTinyMceInNodeRequest($blockType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getUpdateTinyMceInNodeRequest(array $blockType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 7]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to revert the tinymce used in nodes
+     *
+     * @param array blockType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getRevertTinyMceInNodeRequest($blockType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getRevertTinyMceInNodeRequest(array $blockType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 8]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to revert the nodes
+     *
+     * @param array blockType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getRevertNodesRequest($blockType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getRevertNodesRequest(array $blockType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 9]
```diff
@@ -0,0 +1,779 @@
...
+                                for (var j = 0; j < mediaAttribute.length; j++) {
+                                    if (mediaAttribute[j]["id"] != undefined && mediaAttribute[j]["format"] != undefined) {
+                                        mediaAttribute[j] = "' . self::MEDIA_PREFIX . '" + mediaAttribute[j]["id"];
+                                    }
+                                }
+                            }
+                        }';
+        }
+
+        $query .= '
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: same for attributeName

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 10]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to revert the media format in bbcode located in tinymce from contents
+     *
+     * @param array contentType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getRevertTinyMceInContentRequest($contentType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getRevertTinyMceInContentRequest(array $contentType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 11]
```diff
@@ -0,0 +1,779 @@
...
+
+    /**
+     * Get the request to revert the contents
+     *
+     * @param array contentType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getRevertContentsRequest($contentType, $attributes)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php]: protected function getRevertContentsRequest(array $contentType, array $attributes)

File: MediaAdminModelBundle/Migrations/MongoDB/config/media.yml [hunk 12]
```diff
@@ -0,0 +1,30 @@
+contents:
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/config/media.yml]: an integrator can't override this file

**Comment by ngilain** [comment]
@alavieille : updates done

File: MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php [hunk 13]
```diff
@@ -0,0 +1,793 @@
...
+
+    private $container;
+    protected $configuration;
+
+    /**
+     * Set the container
+     *
+     * @param ContainerInterface $container
+     */
+    public function setContainer(ContainerInterface $container = null)
```
Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php:42]: why null in setContainer ?

Comment (ngilain) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php:42]: @alavieille: To respect Symfony\Component\DependencyInjection\ContainerAwareInterface

Comment (alavieille) [inline MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php:42]: ok

**Comment by alavieille** [comment]
:+1:

**Pull request merged**